training.patience = -1 causes immediate stop instead of unlimited stopping

## How to reproduce the behaviour
Not sure if this is maybe a documentation issue, but on the training docs at https://spacy.io/usage/training, inside the source for default_config.cfg is the comment 
```
# Controls early-stopping. 0 or -1 mean unlimited.
patience = 1600
```

I get the expected behaviour when i set patience = 0
```
10:44AM ~/projects/pipelines/textcat_demo> python -m spacy train configs/config.cfg --output training/ --paths.train corpus/train.spacy --paths.dev corpus/dev.spacy --training.patience 0 --training.max_epochs 3
ℹ Using CPU

=========================== Initializing pipeline ===========================
[2021-03-22 10:45:36,285] [INFO] Set up nlp object from config
[2021-03-22 10:45:36,358] [INFO] Pipeline: ['textcat']
[2021-03-22 10:45:36,363] [INFO] Created vocabulary
[2021-03-22 10:45:36,363] [INFO] Finished initializing nlp object
[2021-03-22 10:45:41,542] [INFO] Initialized pipeline components: ['textcat']
✔ Initialized pipeline

============================= Training pipeline =============================
ℹ Pipeline: ['textcat']
ℹ Initial learn rate: 0.001
E    #       LOSS TEXTCAT  CATS_SCORE  SCORE
---  ------  ------------  ----------  ------
  0       0          0.50       41.18    0.41
  0     100         20.82       60.32    0.60
  0     200         13.18       84.83    0.85
  1     300         12.69       84.93    0.85
  1     400          6.19       85.29    0.85
  2     500          3.50       84.65    0.85
  2     600          2.89       85.38    0.85
✔ Saved pipeline to output directory
training/model-last
```
However i get an immediate stop when i set patience = -1
```
10:46AM ~/projects/pipelines/textcat_demo> python -m spacy train configs/config.cfg --output training/ --paths.train corpus/train.spacy --paths.dev corpus/dev.spacy --training.patience -1 --training.max_epochs 3
ℹ Using CPU

=========================== Initializing pipeline ===========================
[2021-03-22 10:48:31,458] [INFO] Set up nlp object from config
[2021-03-22 10:48:31,532] [INFO] Pipeline: ['textcat']
[2021-03-22 10:48:31,537] [INFO] Created vocabulary
[2021-03-22 10:48:31,537] [INFO] Finished initializing nlp object
[2021-03-22 10:48:36,775] [INFO] Initialized pipeline components: ['textcat']
✔ Initialized pipeline

============================= Training pipeline =============================
ℹ Pipeline: ['textcat']
ℹ Initial learn rate: 0.001
E    #       LOSS TEXTCAT  CATS_SCORE  SCORE
---  ------  ------------  ----------  ------
  0       0          0.50       41.18    0.41
✔ Saved pipeline to output directory
training/model-last

```

<!-- Include a code example or the steps that led to the problem. Please try to be as specific as possible. -->

## Your Environment
<!-- Include details of your environment. If you're using spaCy 1.7+, you can also type `python -m spacy info --markdown` and copy-paste the result here.-->
## Info about spaCy

- **spaCy version:** 3.0.5
- **Platform:** Darwin-20.3.0-x86_64-i386-64bit
- **Python version:** 3.7.4
- **Pipelines:** en_core_web_sm (3.0.0)

--- a/spacy/default_config.cfg
+++ b/spacy/default_config.cfg
@@ -68,7 +68,7 @@ seed = ${system.seed}
 gpu_allocator = ${system.gpu_allocator}
 dropout = 0.1
 accumulate_gradient = 1
-# Controls early-stopping. 0 or -1 mean unlimited.
+# Controls early-stopping. 0 disables early stopping.
 patience = 1600
 max_epochs = 0
 max_steps = 20000

